Paintroid/src/test/java/org/catrobat/paintroid/test/presenter/MainActivityPresenterTest.java

```diff
@@ -1007,6 +956,154 @@ public void testOnSaveImagePostExecuteWhenFailedThenShowDialog() {
 		verify(navigator).showSaveErrorDialog();
 	}
 
+	@Test
+	public void testSaveImageClickedWhenPermissionNotGrantedThenAskForPermissionShowInfoDialogIfNotGranted() {
+		doReturn(false).when(navigator).doIHavePermission(Manifest.permission.WRITE_EXTERNAL_STORAGE);
+		doReturn(true).when(navigator).isSdkAboveOrEqualM();
+
+		presenter.saveImageClicked();
+
+		verify(navigator).askForPermission(new String[] {Manifest.permission.WRITE_EXTERNAL_STORAGE},
+				PERMISSION_EXTERNAL_STORAGE_SAVE);
+
+		presenter.handlePermissionRequestResults(PERMISSION_EXTERNAL_STORAGE_SAVE,
```

Comment:
Please stick to the `AAA` pattern for testing.

The `assert` phase should not be succeeded with additionals calls to the presenter.
These tests can be split up in at least two test cases each.